src/CSLOLToolsImpl.cpp

```diff
@@ -277,6 +301,18 @@ void CSLOLToolsImpl::changeIgnorebad(bool ignorebad) {
 }
 
 void CSLOLToolsImpl::init() {
+    #if __APPLE__
+        // 
+        //  TODO:   Wait for user to click “Discard” then exit.
+        // 
+        if (isUnderRosetta() != 0)
+        {
+            doReportError("Invalid architecture", "Cannot run on ARM64, only x86/x64.", "");
+
+            return;
+        }
+    #endif
+
```

Comment:
```suggestion
       auto utils = CSLOLUtils(nullptr);
        if (!utils.isArchSupported()) {
            doReportError("Invalid architecture", "Cannot run on ARM64, only x86/x64.", "");
            setState(CSLOLState::StateIdle);
            return;
        }
```
I would not ifdef this, instead i would uncoditionally return true for non-apple implementation .

Additionally it might make sense to move this into patcher part only here:
https://github.com/LeagueToolkit/cslol-manager/blob/master/src/CSLOLToolsImpl.cpp#L508

Reply:
Why let the application run if the platform isn't supported in the first place?

Reply:
> Why let the application run if the platform isn't supported in the first place?

Because only patching is not supported all the other parts should still work(mod managment, creation, editing etc.)